NitroxServer/Communication/UdpServer.cs

```diff
@@ -36,11 +38,20 @@ public void Start()
 
             Thread thread = new Thread(Listen);
```

Comment:
Can you move this thread init to the constructor? That way you can do `thread.Join()` at the `Stop()` method to wait for the listen thread to exit.